catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/SourcePoller.java

```diff
@@ -13,110 +13,434 @@
+  /**
+   * Sets the {@link FederatedSource}s of which to check the availability periodically according to
+   * {@link #setPollIntervalMinutes(long)}
+   *
+   * @throws IllegalArgumentException if {@code federatedSource} is {@code null}
+   */
+  public void setFederatedSources(final List<FederatedSource> federatedSources) {
+    notNull(federatedSources);
 
-    runner.shutdown();
+    LOGGER.trace(
+        "Setting federated sources: old size={}, new size={}",
+        this.federatedSources.size(),
+        federatedSources.size());
+    this.federatedSources = federatedSources;
+  }
 
-    handle.cancel(true);
+  /**
+   * Sets the {@link CatalogProvider}s of which to check the availability periodically according to
+   * {@link #setPollIntervalMinutes(long)}
+   *
+   * @throws IllegalArgumentException if {@code catalogProvider} is {@code null}
+   */
+  public void setCatalogProviders(final List<CatalogProvider> catalogProviders) {
+    notNull(catalogProviders);
 
-    scheduler.shutdownNow();
+    LOGGER.trace(
+        "Setting catalog providers: old size={}, new size={}",
+        this.catalogProviders.size(),
+        catalogProviders.size());
+    this.catalogProviders = catalogProviders;
   }
 
   /**
-   * Start method for this poller, invoked by the CatalogFramework's blueprint when the framework is
-   * registered/installed. No logic is executed except for logging the framework name.
+   * Sets the {@link CatalogStore}s of which to check the availability periodically according to
+   * {@link #setPollIntervalMinutes(long)}
    *
-   * @param framework the catalog framework being started
-   * @param properties unused, but required by blueprint
+   * @throws IllegalArgumentException if {@code catalogStore} is {@code null}
    */
-  public void start(CatalogFramework framework, Map properties) {
-    LOGGER.debug("Framework started for [{}]", framework);
+  public void setCatalogStores(final List<CatalogStore> catalogStores) {
+    notNull(catalogStores);
+
+    LOGGER.trace(
+        "Setting catalog stores: old size={}, new size={}",
+        this.catalogStores.size(),
+        catalogStores.size());
+    this.catalogStores = catalogStores;
   }
 
-  public int getInterval() {
-    return interval;
+  /**
+   * @param pollIntervalMinutes the interval (in minutes) at which to recheck the availability of
+   *     the {@link Source}s
+   * @throws IllegalArgumentException if {@code pollIntervalMinutes} is less than {@code 1} or is
+   *     greater than {@link Integer#MAX_VALUE}
+   */
+  public void setPollIntervalMinutes(final long pollIntervalMinutes) {
+    final int minimumPollIntervalMinutes = 1;
+    if (pollIntervalMinutes < minimumPollIntervalMinutes) {
+      LOGGER.debug(
+          "pollIntervalMinutes argument may not be less than " + minimumPollIntervalMinutes);
+      throw new IllegalArgumentException(
+          "pollIntervalMinutes argument may not be less than " + minimumPollIntervalMinutes);
+    }
+
```

Comment:
I think we need to cancel the current pollAllSourcesThreadPool future and start a new one based on this poll interval don't we?

Reply:
good catch. I'll see what I can do to implement that.